src/vsr/journal.zig

```diff
@@ -624,7 +624,7 @@ pub fn JournalType(comptime Replica: type, comptime Storage: type) type {
             assert(journal.status == .recovered);
             assert(journal.header_with_op(op_max) != null);
             assert(op_max >= op_min);
-            assert(op_max - op_min + 1 <= slot_count);
+            assert(op_max - op_min <= slot_count);
```

Comment:
I am wondering if we should fix it here, or at the call-site? 

Suppose that we are in a situation where op_min is the checkpoint, and op_max is op_prepare_max. 

I _think_ this means that `op_min` is actually outside our journal --- it is checkpointed, so the journal needs only to include ops higher than that. 

In _this_ function, I think then `const A = journal.header_with_op(op_min);` would return `null`, and we'll return `op_min` as a part of the header break, while it's not the case? 

IOW, I _think_ the caller should be able to assert that any prepare from the resulting range can actually be appended to the journal safely (between op_repair_min and op_prepare_max (🤔 is this a sign that we should rename op_repair_min to op_prepare_min?)). But if we allow op_min=op_checkpoint here, that would be violated?

Reply:
Mhmmmm, thank you for bringing this up, I _think_ you're right about fixing it at the call site... I certainly hadn't thought this one through this clearly and was waiting on VOPR1000 to give me some feedback! Let me read through this function very carefully and dry run this case. 

> But if we allow op_min=op_checkpoint here, that would be violated?
Also let me think about this case as well. 🤔

Reply:
>  I think the caller should be able to assert that any prepare from the resulting range can actually be appended to the journal safely (between op_repair_min and op_prepare_max

Also, this is such a great point, I hadn't thought about it this way at all! Improved my mental model. 😄